gems/core/viewhelpers.py

```diff
@@ -176,4 +180,50 @@ def get_surveyresult_hstore_keys(ui_field=True):
             else:
                 answer_keys.append(answer_key)
 
-    return answer_keys
\ No newline at end of file
+    return answer_keys
+
+
+def process_group_member(api, member, group):
+    # if for some reason we don't have the vumi key in the db for this contact, fetch the contact from vumi
+    if member.vkey is None or member.vkey == '':
+        try:
+            contact = api.get_contact(msisdn=member.msisdn)
+            member.vkey = contact['key']
+            member.save()
+        except Exception:
+            logger.info('Contact: %s not found in vumi' % member)
+            return
+
+    try:
+        api.update_contact(member.vkey, {u'groups': (group.group_key, )})
+    except Exception:
+        logger.info('Contact: %s update failed' % member)
+        return
+
+    group_member, created = ContactGroupMember.objects.get_or_create(group=group, contact=member)
```

Comment:
Add a try/except arround the get_or_create and save to make sure the job doesn't stop halfway